plugins/handlers/gallery.py

```diff
@@ -351,56 +434,40 @@ def metadata_for_media_file(root, path, title_from_filename):
 
-    # TODO: Support specifying image size sets in the configuration file #10
-
-    # Determine which profiles to use; we use a different profile for equirectangular projections.
-    profiles = DEFAULT_PROFILES
-    if "projection" in metadata and metadata["projection"] == "equirectangular":
-        profiles = EQUIRECTANGULAR_PROFILES
-
-    # Generate the various different image sizes.
-    for profile_name, profile_details in profiles.items():
-
-        filename = "%s-%s%s" % (identifier, profile_name.replace("_", "-"), ext)
-        if profile_name == "image":
-            filename = "%s%s" % (identifier, ext)
-
-        scale = 1
-        if "scale" in profile_details:
-            scale = profile_details["scale"]
-        width = None
-        if "width" in profile_details:
-            width = profile_details["width"]
-        height = None
-        if "height" in profile_details:
-            height = profile_details["height"]
-
-        source_scale = metadata["scale"] if metadata["scale"] is not None else 1
-
-        # TODO: Rename generate_thumbnail.
-        metadata[profile_name] = generate_thumbnail(source_path,
-                                                    destination, dirname, filename,
-                                                    (width, height),
-                                                    source_scale, scale)  # TODO: Why two scales?
+    name, _ = os.path.splitext(basename)
+    name = name.lower()
+    destination_path = os.path.join(destination, dirname, name)
+    utils.makedirs(destination_path)
+    transform_files = []
+    transform_metadata = collections.defaultdict(list)
+    transforms = evaluate_tests(IMAGE_TRANSFORMS, {"basename": os.path.basename(source_path),
+                                                   "metadata": metadata})
+    for transform in transforms:
+        result = transform.perform(path=source_path, destination=destination_path)
+        if not isinstance(result, TransformResult):
+            raise AssertionError("Invalid result type.")
+        transform_files.extend(result.files)
+        transform_documents = []
+        for document in result.documents:
+            document["filename"] = os.path.join(name, document["filename"])
+            document["url"] = os.path.join("/", dirname, name) + document["url"]
+            transform_documents.append(document)
+        for tag in transform.tags:
+            transform_metadata[tag].extend(transform_documents)
 
-    # Configure the page details.
     metadata["category"] = category
     metadata["template"] = "photo.html"  # TODO: Pass this in as a default.
     metadata["path"] = converters.ensure_leading_slash(os.path.join(dirname, basename))
+    for key, value in transform_metadata.items():
+        metadata[key] = value if len(value) > 1 else value[0]  # Single arrays should be direct values.
+    if metadata["scale"] is None:  # Ensure images have a scale.
+        metadata["scale"] = 1
```

Comment:
This should probably be done in the parent utility that does this (and tested appropriately).